### PR TITLE
feat: migrate to Algolia

### PR DESCRIPTION
**Pull request by neo773**
/claim #167

### CODE REVIEW

**Comment by tusharmath** [comment]
@neo773 How are we doing on this?

**Comment by amitksingh1490** [comment]
@neo773  can you resolve the conflicts once and attach a video of this working

File: src/components/shared/Search/Icons.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from "react"
+
+const LvlIcon: React.FC = () => {
+  return (
```
Comment (amitksingh1490) [inline src/components/shared/Search/Icons.tsx]: can we avoid inlining svgs

File: src/components/shared/Search/Results.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            >
+              <SourceIcon type={item.type} />
+              <div className="flex flex-col">
+                <div
+                  className="text-[14px]"
+                  dangerouslySetInnerHTML={{__html: item?._snippetResult?.hierarchy?.lvl2?.value}}
+                ></div>
+                <div
+                  className="text-[14px]"
+                  dangerouslySetInnerHTML={{__html: item?._snippetResult?.hierarchy?.lvl1?.value}}
```
Comment (amitksingh1490) [inline src/components/shared/Search/Results.tsx]: avoid dangerouslySetInnerHTML

File: src/components/shared/Search/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+import {DocSearchHit} from "@docsearch/react/dist/esm/types"
+import {performSearch} from "./utils"
+import SearchInput from "./SearchBar"
+import Results from "./Results"
+
+const algoliaConfig = {
+  appId: "X27WDVHRQ3",
+  apiKey: "35bc100f239853cd8a7195b23ed7393b",
+  indexName: "tailcall",
+}
```
Comment (amitksingh1490) [inline src/components/shared/Search/Search.tsx]: env variables

**Pull request closed without merging**